EXPRESSIONS.md

```diff
@@ -0,0 +1,80 @@
+# Example WDL for CancerExomePipeline_v2
+
+```
+task M2 {
+  File ref_fasta
+  File ref_fasta_dict
+  File ref_fasta_fai
+  File tumor_bam
+  File tumor_bai
+  File normal_bam
+  File normal_bai
+  File intervals
+  String m2_output_vcf_name
+
+  command {
+    java -jar /task/GenomeAnalysisTK_latest_unstable.jar -T M2 \
+    --no_cmdline_in_header -dt NONE -ip 50 \
+    -R ${ref_fasta} \
+    -I:tumor ${tumor_bam} \
+    -I:normal ${normal_bam} \
+    -L ${intervals} \
+    -o ${m2_output_vcf_name} \
+    --disable_auto_index_creation_and_locking_when_reading_rods
+  }
+  runtime {
+    docker: "gcr.io/broad-dsde-dev/cep3"
+  }
+  output {
+    File m2_output_vcf = "${m2_output_vcf_name}"
+  }
+}
+
+workflow CancerExomePipeline_v2 {
+  call M2
+}
+```
+# Example Root Entity
+
+Entity Name: ```HCC1143_pair```
+
+Entity Type: ```pair```
+
+Entity Attributes: ```control_bai```, ```case_bai```, ```output_vcf```, ```ref_pac```, ```ref_ann```, ```case_bam```, ```case_sample```, ```participant```, ```vcf_output_name```, ```ref_fasta```, ```ref_amb```, ```ref_intervals```, ```control_sample```, ```ref_sa```, ```ref_bwt```, ```control_bam```, ```ref_fai```, ```ref_dict```
+
+# Background
+
+An entity can be a sample, participant, pair, or a set of any of the aforementioned types. Entities have properties called "attributes", which may store values such as file paths or references to other entities. Workspaces also have attributes. These attributes can be configured to workflow inputs and outputs using method configurations.
+
+Input and output expressions will tell the configuration where to find the inputs for the method and where to store the outputs of the method. When writing an expression, ```this``` refers to the root entity that is selected when configuring a method. To write an expression that binds to data associated with the root entity, use the ```this.<attribute_name>``` syntax (i.e. to access a reference FASTA associated with the example root entity (shown above), use ```this.ref_fasta```, where ```ref_fasta``` is the attribute name associated with your FASTA file pointer.) To bind data in the workspace, use the ```workspace.<attribute_name>``` syntax, which functions similarly to the ```this.``` syntax.
```

Comment:
inputs can dereference intermediate entities, too: this.(some_other_entity.)*attribute. outputs can't.